app/models/order.rb

```diff
@@ -297,15 +309,11 @@ def self.join_order_associations
   end
 
   def self.recently_used(user_id)
-    active_orders
-    .select("DISTINCT ON (orders.id) orders.id AS key,  versions.created_at AS recently_used_at").
-    joins("INNER JOIN versions ON ((object_changes -> 'order_id' ->> 1) = CAST(orders.id AS TEXT))").
-    joins("INNER JOIN packages ON (packages.id = versions.item_id AND versions.item_type = 'Package')").
-    where(" versions.event = 'update' AND
-      (object_changes ->> 'order_id') IS NOT NULL AND
-      CAST(whodunnit AS integer) = ? AND
-      versions.created_at >= ? ", user_id, 15.days.ago).
-    order("key, recently_used_at DESC")
+    active_orders.goodcity_orders.joins(:versions).where(versions: {id: Version.select(:id).where(whodunnit: user_id.to_s, item_type: "Order")}).distinct.order(updated_at: :desc).limit(5)
```

Comment:
this line is too long and hard to read... can you shorten. also, what is the performance of this compare to the previous code?

Reply:
Previous query was not giving expected result and it even was not working(I suspect). I tried to add change in that query itself but it was difficult to debug. That is why I rewrite the query again. I'll look into improve and shorten the query.